lib/Dialect/Triton/Transforms/Combine.cpp

```diff
@@ -18,35 +17,7 @@ namespace mlir::triton {
 namespace {
 
 bool isZero(Value val) {
-  if (matchPattern(val, m_Zero()) || matchPattern(val, m_AnyZeroFloat()))
-    return true;
-  // broadcast(constant_0)
-  if (auto bc = val.getDefiningOp<BroadcastOp>()) {
```

Comment:
Since the input is now canonicalized, we don't need to match against non-canonical forms.